src/secp256k1/src/modules/multiset/tests_impl.h

```diff
@@ -0,0 +1,326 @@
+    secp256k1_multiset_add(ctx, &r3, elements[1], DATALEN);
+    secp256k1_multiset_combine(ctx, &r2, &r3);
+    CHECK_EQUAL(&r1,&r2); /* M(0,1,2) == M(2,0)+M(1) */
+
+    secp256k1_multiset_combine(ctx, &r2, &empty);
+    CHECK_EQUAL(&r1,&r2); /* M(0,1,2)+M() == M(0,1,2) */
+    secp256k1_multiset_combine(ctx, &r2, &r1);
+    CHECK_NOTEQUAL(&r1,&r2); /* M(0,1,2)+M(0,1,2) != M(0,1,2) */
+}
+
+
+void test_remove(void) {
+
+    /* Testing removal of elements */
+    secp256k1_multiset empty, r1,r2,r3;
+
+    secp256k1_multiset_init(ctx, &empty);
+    secp256k1_multiset_init(ctx, &r1);
+    secp256k1_multiset_init(ctx, &r2);
+    secp256k1_multiset_init(ctx, &r3);
+
+    CHECK_EQUAL(&r1,&r2); /* M()==M() */
+
+    secp256k1_multiset_add   (ctx, &r1, elements[0], DATALEN);
+    secp256k1_multiset_add   (ctx, &r1, elements[1], DATALEN);
+    secp256k1_multiset_add   (ctx, &r1, elements[3], DATALEN);
+    secp256k1_multiset_add   (ctx, &r1, elements[9], DATALEN);
+    secp256k1_multiset_add   (ctx, &r1, elements[8], DATALEN);
+
+    secp256k1_multiset_add   (ctx, &r2, elements[1], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[9], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[11], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[10], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[0], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[10], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[3], DATALEN);
+    secp256k1_multiset_add   (ctx, &r2, elements[8], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[11], DATALEN);
+
+    secp256k1_multiset_add   (ctx, &r3, elements[9], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[15], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[15], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[1], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[9], DATALEN);
+    secp256k1_multiset_remove(ctx, &r3, elements[15], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[0], DATALEN);
+    secp256k1_multiset_remove(ctx, &r3, elements[15], DATALEN);
+    secp256k1_multiset_remove(ctx, &r3, elements[9], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[3], DATALEN);
+    secp256k1_multiset_add   (ctx, &r3, elements[8], DATALEN);
+
+    CHECK_EQUAL(&r1,&r2); /* M(0,1,3,9,8)==M(1,9,11,10,9,3,8)-M(10,11) */
+    CHECK_EQUAL(&r1,&r3); /* M(0,1,3,9,8)==M(9,15,15,1,9,0,3,8)-M(15,15,9) */
+    CHECK_NOTEQUAL(&r1,&empty); /* M(0,1,3,9,8)!=M() */
+
+    secp256k1_multiset_remove(ctx, &r3, elements[8], DATALEN);
+    CHECK_NOTEQUAL(&r1,&r3); /* M(0,1,3,9,8)-M(8)!=M(0,1,3,9,8) */
+
+    secp256k1_multiset_remove(ctx, &r2, elements[0], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[9], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[8], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[1], DATALEN);
+    secp256k1_multiset_remove(ctx, &r2, elements[3], DATALEN);
```

Comment:
Valgrind is unhappy here:
```
==22704== Conditional jump or move depends on uninitialised value(s)
==22704==    at 0x10D254: secp256k1_fe_normalize (field_5x52_impl.h:83)
==22704==    by 0x11CA2D: multiset_add_remove.constprop.70 (main_impl.h:111)
==22704==    by 0x13AF00: secp256k1_multiset_remove (main_impl.h:126)
==22704==    by 0x13AF00: test_remove (tests_impl.h:200)
==22704==    by 0x13BC8F: run_multiset_tests (tests_impl.h:320)
==22704==    by 0x1094AB: main (tests.c:5073)
```

Reply:
I think valgrind is unhappy with gej.x,y,z being undefined when gej.infinite. I cleared the target to fix this.